erpnext/stock/doctype/serial_no/serial_no.py

```diff
@@ -554,7 +554,7 @@ def auto_fetch_serial_number(qty, item_code, warehouse, posting_date=None, batch
 
 	if batch_nos:
 		try:
-			filters["batch_no"] = json.loads(batch_nos)
+			filters["batch_no"] = [json.loads(batch_nos)]
```

Comment:
We should check if `batch_nos` is a list, if not then wrap it around a `[]`

This might break if `batch_nos` is already passed as a list